libsolidity/ast/Types.cpp

```diff
@@ -406,28 +603,52 @@ bool IntegerConstantType::isImplicitlyConvertibleTo(Type const& _convertTo) cons
 			return true;
 		return false;
 	}
+	else if (_convertTo.category() == Category::FixedPoint)
+	{
+		auto targetType = dynamic_cast<FixedPointType const*>(&_convertTo);
+		if (m_value == 0)
+			return true;
+		int forSignBit = (targetType->isSigned() ? 1 : 0);
+		if (m_value > 0)
+		{
+			bool properlyScaledBits = m_scalingFactor <= targetType->fractionalBits() ? 
+				true : m_scalingFactor == 1 && targetType->fractionalBits() == 0 ? true : false;
+			if (m_value <= (u256(-1) >> (256 - targetType->numBits() + forSignBit)) && properlyScaledBits)
+				return true;
+			else if (targetType->isSigned() && -m_value <= (u256(1) << (targetType->numBits() - forSignBit)) && properlyScaledBits)
+				return true;
+			return false;
+		}
+
+	}
 	else if (_convertTo.category() == Category::FixedBytes)
 	{
 		FixedBytesType const& fixedBytes = dynamic_cast<FixedBytesType const&>(_convertTo);
 		return fixedBytes.numBytes() * 8 >= integerType()->numBits();
 	}
-	else
-		return false;
+	return false;
 }
 
-bool IntegerConstantType::isExplicitlyConvertibleTo(Type const& _convertTo) const
+bool ConstantNumberType::isExplicitlyConvertibleTo(Type const& _convertTo) const
 {
-	TypePointer intType = integerType();
-	return intType && intType->isExplicitlyConvertibleTo(_convertTo);
+	if (m_value.denominator() == 1)
+	{
+		TypePointer intType = integerType();
+		return intType && intType->isExplicitlyConvertibleTo(_convertTo);
+	}
+	TypePointer fixType = fixedPointType();
```

Comment:
I think we have to do that more manually here, because `1/3` does not have a canonical fixed point type, but `fixed(1/3)` is still a valid explicit conversion.